test/cppsuite/tests/search_near_02.cxx

```diff
@@ -274,69 +274,47 @@ class search_near_02 : public test_harness::test {
-            }
+            testutil_assert(exact_prefix == 1);
 
-            /* We should be at the key pointed by the prefix cursor. */
-            testutil_assert(ret != WT_NOTFOUND && k_str == key_prefix_str);
+            /*
+             * The next key of the default cursor should be equal to the key pointed by the prefix
+             * cursor.
+             */
+            testutil_assert(cursor_default->next(cursor_default.get()) == 0);
+            testutil_check(cursor_default->get_key(cursor_default.get(), &k));
+            testutil_assert(k == key_prefix_str);
         }
         /* Example: */
         /* keys: a, bb, bba */
+        /* All keys are visible. */
         /* Default search_near(bb) returns bb, exact = 0 */
         /* Prefix search_near(bb) returns bb, exact = 0 */
-        /* Prefix search_near(bb) returns bba, exact = 1 */
         else if (exact_default == 0) {
-            /* The key at the default cursor should be the prefix. */
-            testutil_assert(key_default_str == prefix);
-
-            if (exact_prefix == 0) {
-                /* Both cursors should be pointing at the same key. */
-                testutil_assert(key_default_str == key_prefix_str);
-            } else {
-                /*
-                 * The next key of the default cursor should be equal to the key pointed by the
-                 * prefix cursor.
-                 */
-                testutil_assert(cursor_default->next(cursor_default.get()) == 0);
-                testutil_check(cursor_default->get_key(cursor_default.get(), &k));
-                testutil_assert(k == key_prefix_str);
-            }
+            /* Both cursors should be pointing at the same key which is the prefix. */
+            testutil_assert(exact_prefix == exact_default);
+            testutil_assert(key_default_str == prefix && key_prefix_str == prefix);
         }
         /* Example: */
         /* keys: a, bb, bba */
-        /* Default search_near(bb) returns bba, exact = 1 */
-        /* Prefix search_near(bb) returns bb, exact = 0 */
-        /* Prefix search_near(bb) returns bba, exact = 1 */
+        /* Only bb is not visible. */
+        /* Default search_near(bb) returns bba, exact = 1. */
+        /* Prefix search_near(bb) returns bba, exact = 1. */
         else {
-            if (exact_prefix == 0) {
-                /*
-                 * The previous key of the default cursor should be equal to the key pointed by the
-                 * prefix cursor.
-                 */
-                testutil_assert(cursor_default->prev(cursor_default.get()) == 0);
-                testutil_check(cursor_default->get_key(cursor_default.get(), &k));
-                testutil_assert(k == key_prefix_str);
-            } else {
-                /* Both cursors should be pointing at the same key. */
-                testutil_assert(key_default_str == key_prefix_str);
-            }
+            /* Both cursors should be pointing at the same key. */
```

Comment:
In this case it is actually possible for default search near to return back -1. Imagine a case where we have keys aaz, aba and both are visible to us. If we perform default search near on ab, it is possible that we return back aaz. While prefix search near returns back 1.